meta-client/src/main/java/org/apache/helix/metaclient/api/MetaClientInterface.java

```diff
@@ -0,0 +1,158 @@
+package org.apache.helix.metaclient.api;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+
+
+public interface MetaClientInterface<T> {
+
+  enum EntryMode {
+    //The node will be removed automatically when the session associated with the creation
+    // of the node expires.
+    EPHEMERAL,
+    //The node will not be automatically deleted upon client's disconnect.
+    PERSISTENT
```

Comment:
@xyuanlu Are we not supporting "SEQUENTIAL" in basic CRUD metaclient interface? Or you want to add it in only zk specific implementation?

Reply:
We plan to add it later.